test/test.js

```diff
@@ -2309,6 +2309,70 @@ exports.take = {
     'noValueOnError': noValueOnErrorTest(_.take(1)),
 };
 
+exports.takeWhile = {
+    'arrayStream': function (test) {
```

Comment:
Add a test to make sure that the source stream isn't fully consumed. You can use a generator stream for this

```javascript
var i = 0;
_(function (push, next) {
  push(null, i++);
  if (i < 10) {
    push(null, _.nil);
  } else {
    next();
  }
}).takeWhile(function (x) { return x < 3; })
.done(function () {
  // check that i == 5.
});
```

Reply:
done!